src/Features/Core/Portable/NavigationBar/INavigationBarItemService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.NavigationBar
+{
+    internal interface INavigationBarItemService : ILanguageService
+    {
+        Task<ImmutableArray<RoslynNavigationBarItem>> GetItemsAsync(Document document, bool supportsCodeGeneration, CancellationToken cancellationToken);
```

Comment:
core type is pure-data only.  I need to pass in `supportsCodeGeneration` because the workspace on the OOP side says that it doesn't support changes, and i didn't want to have to figure out what needs to be updated in teh workspace+oop to get that answering the same way in OOP as in VS.